app/models/dataset/Dataset.scala

```diff
@@ -733,6 +762,53 @@ class DatasetMagsDAO @Inject()(sqlClient: SqlClient)(implicit ec: ExecutionConte
     replaceSequentiallyAsTransaction(clearQuery, insertQueries)
   }
 
+  def updateMagPathsForDataset(datasetId: ObjectId, magPaths: List[MagPathInfo]): Fox[Unit] =
+    for {
+      _ <- Fox.successful(())
+      updateQueries = magPaths.map(pathInfo => {
+        val magLiteral = s"(${pathInfo.mag.x}, ${pathInfo.mag.y}, ${pathInfo.mag.z})"
+        q"""UPDATE webknossos.dataset_mags
+                 SET path = ${pathInfo.path}, realPath = ${pathInfo.realPath}, haslocaldata = ${pathInfo.hasLocalData}
+                 WHERE _dataset = $datasetId
+                  AND datalayername = ${pathInfo.layerName}
```

Comment:
please follow capitalization from schema.sql here, dataLayerName and hasLocalData